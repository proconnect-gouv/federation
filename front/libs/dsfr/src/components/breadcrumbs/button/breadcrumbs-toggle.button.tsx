import { useToggle } from 'usehooks-ts';

import { t } from '@fc/i18n';

interface BreadCrumbsToggleButtonProps {
  id: string;
}

export const BreadCrumbsToggleButton = ({ id }: BreadCrumbsToggleButtonProps) => {
  const [show, toggleShow] = useToggle(false);

  return (
    <button
      aria-controls={id}
      aria-expanded={show}
      className="fr-breadcrumb__button"
      onClick={toggleShow}>
      {t('DSFR.breadcrumbs.show')}
    </button>
  );
};

BreadCrumbsToggleButton.displayName = 'BreadCrumbsToggleButton';
