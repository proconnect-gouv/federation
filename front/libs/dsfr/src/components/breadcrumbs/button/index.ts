export * from './breadcrumbs-toggle.button';
