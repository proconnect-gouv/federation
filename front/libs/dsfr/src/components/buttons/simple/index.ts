/* istanbul ignore file */

// declarative file
export * from './simple.button';
