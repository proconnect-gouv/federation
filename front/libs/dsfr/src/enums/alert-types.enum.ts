/* istanbul ignore file */
export enum AlertTypes {
  ERROR = 'error',
  INFO = 'info',
  SUCCESS = 'success',
  WARNING = 'warning',
}
