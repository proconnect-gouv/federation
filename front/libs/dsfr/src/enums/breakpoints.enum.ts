/* istanbul ignore file */
export enum Breakpoints {
  XS = 576,
  SM = 768,
  MD = 992,
  LG = 1248,
}
