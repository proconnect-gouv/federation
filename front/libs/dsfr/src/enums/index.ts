/* istanbul ignore file */

// declarative file
export * from './alert-types.enum';
export * from './button-types.enum';
export * from './connect-types.enum';
export * from './icon-placement.enum';
export * from './options.enum';
export * from './pagination.enum';
export * from './priorities.enum';
export * from './sizes.enum';
