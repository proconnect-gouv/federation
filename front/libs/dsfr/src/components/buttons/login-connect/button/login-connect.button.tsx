import classnames from 'classnames';
import React from 'react';

import type { PropsWithOnClick } from '@fc/common';
import { t } from '@fc/i18n';

import type { ButtonTypes } from '../../../../enums';
import { ConnectTypes } from '../../../../enums';
import styles from './login-connect.button.module.scss';

interface LoginConnectButtonProps extends PropsWithOnClick {
  type: ButtonTypes;
  connectType: ConnectTypes;
}

export const LoginConnectButton = React.memo(
  ({ connectType, onClick, type }: LoginConnectButtonProps) => {
    const loginLabel = t('DSFR.button.loginConnect.authenticateWith');
    return (
      <button
        className={classnames('fr-connect', {
          [styles['fr-connect--proconnect']]: connectType === ConnectTypes.PRO_CONNECT,
        })}
        type={type}
        onClick={onClick}>
        <span className="fr-connect__login">{loginLabel}</span>
        <span className="fr-connect__brand">{connectType}</span>
      </button>
    );
  },
);

LoginConnectButton.displayName = 'LoginConnectButton';
