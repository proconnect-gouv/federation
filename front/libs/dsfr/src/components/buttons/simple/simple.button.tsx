import classnames from 'classnames';
import type { MouseEventHandler } from 'react';
import React from 'react';

import { ButtonTypes, IconPlacement, Priorities, Sizes } from '../../../enums';
import type { ButtonInterface } from '../../../interfaces/button.interface';

export interface SimpleButtonProps extends ButtonInterface {
  title?: string | undefined;
  type?: ButtonTypes | undefined;
  disabled?: boolean | undefined;
  onClick?: MouseEventHandler<HTMLButtonElement> | undefined;
}

export const SimpleButton = React.memo(
  ({
    children: label,
    className = undefined,
    dataTestId = undefined,
    disabled = undefined,
    icon = undefined,
    iconPlacement = IconPlacement.RIGHT,
    noOutline = false,
    onClick = undefined,
    priority = Priorities.PRIMARY,
    size = Sizes.MEDIUM,
    title = undefined,
    type: nativeButtonType = ButtonTypes.BUTTON,
  }: SimpleButtonProps) => (
    <button
      className={classnames(
        `fr-btn fr-btn--${size}`,
        {
          // DSFR classname
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'fr-btn--secondary': priority === Priorities.SECONDARY,
          // DSFR classname
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'fr-btn--tertiary': priority === Priorities.TERTIARY && !noOutline,
          // DSFR classname
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'fr-btn--tertiary-no-outline': priority === Priorities.TERTIARY && noOutline,
          [`fr-btn--icon-${iconPlacement}`]: !!icon,
          [`fr-icon-${icon}`]: !!icon,
        },
        className,
      )}
      data-testid={dataTestId}
      disabled={disabled}
      title={title}
      type={nativeButtonType}
      onClick={onClick}>
      {label}
    </button>
  ),
);

SimpleButton.displayName = 'SimpleButton';
