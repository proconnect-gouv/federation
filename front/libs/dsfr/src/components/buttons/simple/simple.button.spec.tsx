import { fireEvent, render } from '@testing-library/react';

import { ButtonTypes, IconPlacement, Priorities, Sizes } from '../../../enums';
import { SimpleButton } from './simple.button';

describe('SimpleButton', () => {
  it('should match the snapshot', () => {
    // When
    const { container } = render(<SimpleButton>any-label-mock</SimpleButton>);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, when label is hidden', () => {
    // When
    const { container } = render(
      <SimpleButton hideLabel icon="any-icon-mock" iconPlacement={'left' as IconPlacement} />,
    );

    // Then
    expect(container).toMatchSnapshot();
    expect(container).not.toHaveClass('fr-btn--icon-left');
  });

  it('should match the snapshot, when size is defined to Sizes.LARGE', () => {
    // When
    const { container } = render(<SimpleButton size={Sizes.LARGE}>any-label-mock</SimpleButton>);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, when size is defined to Sizes.MEDIUM', () => {
    // When
    const { container } = render(<SimpleButton size={Sizes.MEDIUM}>any-label-mock</SimpleButton>);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, when size is defined to Sizes.SMALL', () => {
    // When
    const { container } = render(<SimpleButton size={Sizes.SMALL}>any-label-mock</SimpleButton>);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should render the firstchild as a button', () => {
    // When
    const { container, getByRole } = render(<SimpleButton>any-label-mock</SimpleButton>);
    const element = getByRole('button');

    // Then
    expect(element).toStrictEqual(container.firstChild);
  });

  it('should render the label', () => {
    // When
    const { getByText } = render(<SimpleButton>any-label-mock</SimpleButton>);
    const linkElement = getByText('any-label-mock');

    // Then
    expect(linkElement).toBeInTheDocument();
  });

  it('should have the disable attribute', () => {
    // When
    const { getByRole } = render(<SimpleButton disabled>any-label-mock</SimpleButton>);
    const element = getByRole('button');

    // Then
    expect(element).toHaveAttribute('disabled');
  });

  it('should have the title attribute', () => {
    // When
    const { getByRole } = render(
      <SimpleButton title="any-title-mock">any-label-mock</SimpleButton>,
    );
    const element = getByRole('button');

    // Then
    expect(element).toHaveAttribute('title', 'any-title-mock');
  });

  it('should have the type attribute defined', () => {
    // When
    const { getByRole } = render(
      <SimpleButton type={ButtonTypes.RESET}>any-label-mock</SimpleButton>,
    );
    const element = getByRole('button');

    // Then
    expect(element).toHaveAttribute('type', 'reset');
  });

  it('should call the callback mock when user click the button', () => {
    // Given
    const onClickMock = jest.fn();
    // When
    const { getByRole } = render(<SimpleButton onClick={onClickMock}>any-label-mock</SimpleButton>);
    const element = getByRole('button');
    fireEvent.click(element);

    // Then
    expect(onClickMock).toHaveBeenCalledOnce();
  });

  it('should have the classname when priority is equal to Priorities.SECONDARY', () => {
    // When
    const { getByRole } = render(
      <SimpleButton priority={Priorities.SECONDARY}>any-label-mock</SimpleButton>,
    );
    const element = getByRole('button');

    // Then
    expect(element).toHaveClass('fr-btn--secondary');
  });

  it('should have the classname when priority is equal to Priorities.TERTIARY', () => {
    // When
    const { getByRole } = render(
      <SimpleButton priority={Priorities.TERTIARY}>any-label-mock</SimpleButton>,
    );
    const element = getByRole('button');

    // Then
    expect(element).toHaveClass('fr-btn--tertiary');
  });

  it('should have the classname when priority is equal to Priorities.TERTIARY and noOutline is defined', () => {
    // When
    const { getByRole } = render(
      <SimpleButton noOutline priority={Priorities.TERTIARY}>
        any-label-mock
      </SimpleButton>,
    );
    const element = getByRole('button');

    // Then
    expect(element).toHaveClass('fr-btn--tertiary-no-outline');
  });

  it('should have the classname when icon is defined', () => {
    // When
    const { getByRole } = render(<SimpleButton icon="any-icon-mock">any-label-mock</SimpleButton>);
    const element = getByRole('button');

    // Then
    expect(element).toHaveClass('fr-icon-any-icon-mock');
    expect(element).toHaveClass('fr-btn--icon-right');
  });

  it('should have the classname when icon placement is defined', () => {
    // When
    const { getByRole } = render(
      <SimpleButton icon="any-icon-mock" iconPlacement={IconPlacement.LEFT}>
        any-label-mock
      </SimpleButton>,
    );
    const element = getByRole('button');

    // Then
    expect(element).toHaveClass('fr-icon-any-icon-mock');
    expect(element).toHaveClass('fr-btn--icon-left');
  });

  it('should have a custom classname when props is defined', () => {
    // When
    const { getByRole } = render(
      <SimpleButton
        className="any-classname-mock"
        icon="any-icon-mock"
        iconPlacement={IconPlacement.LEFT}>
        any-label-mock
      </SimpleButton>,
    );
    const element = getByRole('button');

    // Then
    expect(element).toHaveClass('any-classname-mock');
  });

  it('should have a custom data-testid when props is defined', () => {
    // When
    const { getByTestId } = render(
      <SimpleButton
        className="any-classname-mock"
        dataTestId="any-datatestid-mock"
        icon="any-icon-mock"
        iconPlacement={IconPlacement.LEFT}>
        any-label-mock
      </SimpleButton>,
    );
    const element = getByTestId('any-datatestid-mock');

    // Then
    expect(element).toBeInTheDocument();
  });
});
