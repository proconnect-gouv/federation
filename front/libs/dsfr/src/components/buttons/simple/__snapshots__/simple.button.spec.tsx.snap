// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleButton should match the snapshot 1`] = `
<div>
  <button
    class="fr-btn fr-btn--md"
  >
    any-label-mock
  </button>
</div>
`;

exports[`SimpleButton should match the snapshot, when size is defined to Sizes.LARGE 1`] = `
<div>
  <button
    class="fr-btn fr-btn--lg"
  >
    any-label-mock
  </button>
</div>
`;

exports[`SimpleButton should match the snapshot, when size is defined to Sizes.MEDIUM 1`] = `
<div>
  <button
    class="fr-btn fr-btn--md"
  >
    any-label-mock
  </button>
</div>
`;

exports[`SimpleButton should match the snapshot, when size is defined to Sizes.SMALL 1`] = `
<div>
  <button
    class="fr-btn fr-btn--sm"
  >
    any-label-mock
  </button>
</div>
`;
